hyperspy/axes.py

```diff
@@ -2001,18 +2011,16 @@ def update_axes_attributes_from(self, axes,
         if changes:
             self.events.any_axis_changed.trigger(obj=self)
 
-    def _update_attributes(self):
+    @observe("_axes")
+    @observe(trait("_axes", notify=False).list_items())
```

Comment:
Just to make sure that I am following correctly, `@observe(trait("_axes", notify=False).list_items())` is to call the `add_axis_manger` method when the attribute, which are of `traits` instances, are changed? If so, I don't see why we need it here? The `axes_manager` attribute of the axis needs to be set  only when the axis is added to a `AxesManager` (currently covered by `@observe("_axes")`)?

Reply:
So the first case is triggered by someone setting the axes value. Something like `am.axes=[axis1, axis2]`

Because lists are mutable in place though that observation isn't triggered by a inplace operation. For that we need the second observation which triggers when something inplace occurs. Ie. `am.axes[1]= axis1`


See https://docs.enthought.com/traits/traits_user_manual/notification.html#notification-handler for more details.